woodchipper/http/fastapi.py

```diff
@@ -1,80 +1,84 @@
 import uuid
-from typing import Any, Callable, Coroutine
+from typing import Callable
 
-from fastapi import FastAPI, Request, Response
+from fastapi import FastAPI, Request
+from starlette.types import ASGIApp, Receive, Scope, Send
 
 from woodchipper.context import LoggingContext, logging_ctx
 
 BLACKLISTED_HEADERS = ["authorization", "cookie"]
 
 
 class WoodchipperFastAPI:
-    def __init__(self, app: FastAPI, blacklisted_headers=BLACKLISTED_HEADERS, request_id_factory=None):
+    def __init__(
+        self,
+        app: FastAPI,
+        request_id_factory=None,
+        blacklisted_headers=BLACKLISTED_HEADERS,
+    ):
+        super().__init__()
         self._app = app
         self._blacklisted_headers = blacklisted_headers
         self._request_id_factory = request_id_factory or (lambda: str(uuid.uuid4()))
 
-    def get_custom_route_handler(self, gen_id, blacklisted_headers):
-        def get_route_handler(self) -> Callable[[Request], Coroutine[Any, Any, Response]]:
+    def _wrap_build_middleware_stack(self, original_fn: Callable) -> Callable:
+        def __wrapped__() -> ASGIApp:
```

Comment:
I was confused by this myself, but seemingly no-- and when I attempted to supply a self argument, I got an error that stating that the positional argument is missing. I couldn't find docs to explain why this might be, but I tested it in TAS and this was the only way I found it to work.